scripts/allowlist/cli/allowlist.js

```diff
@@ -0,0 +1,228 @@
+        return "0x" + str;
+    } else if (!need0x && str.startsWith("0x")) {
+        return str.substr(2);
+    }
+    return str;
+}
+
+function getHex(str, len, need0x, name) {
+    var re = XRegExp(`^0x[0-9A-Fa-f]{${len}}$`);
+    if (!re.test(prefix0x(true,str))) {
+        console.log(`ERROR: Invalid hex string for ${name}: ${str}`);
+        if (str.length !== len+2) {
+            console.log(`Expected length is ${len} digits, actual length is ${str.length-2} digits`)
+        }
+        process.exit(-1);
+    }
+    return prefix0x(need0x, str);
+}
+
+function validHexAccount(str) {
+    var re = XRegExp(`^0x[0-9A-Fa-f]{40}$`);
+    if (!re.test(str)) {
+        if (str.length !== 42) {
+            return false;
+        }
+    }
+    return true;
+}
+
+async function getEvent(eventname, mycontract, receipt) {
+    const event = await mycontract.getPastEvents(eventname,
+        {fromBlock: receipt.blockNumber, toBlock: receipt.blockNumber});
+    console.log(`Event from transaction:\n${JSON.stringify(event[0], null, 4)}`)
+}
+
+async function doWork() {
+    const myAccount = web3.eth.accounts.privateKeyToAccount(prefix0x(true, argv.privateKey));
+    const mycontract = await new web3.eth.Contract(contractJson, getHex(argv.contractAddress, 40, false, "contractAddress"));
+
+    let validators;
+    // check whether the contract address is correct
+    try {
+        validators = await mycontract.methods.getValidators().call();
+        if (validators.length < 1 || !validHexAccount(validators[0])) {
+            console.log("The contract address provided is not correct. Please check and rerun!");
+            console.log(`Got the following return value calling the getValidators method on the contract:\n${validators}`)
+            process.exit(-1);
+        }
+    } catch(e) {
+        console.log("The contract address provided is not correct. Please check and rerun!");
+        console.log(`Got the following error message calling the getValidators method on the contract:\n${e.message}`)
+        process.exit(-1);
+    }
+
+    let success;
+    let status;
+    let receipt;
+    try {
+        switch (argv._[0]) {
+            case "getValidators":
+                // const validators = await mycontract.methods.getValidators().call();
+                console.log(`Validators: ${validators}`);
+                break;
+            case "numAllowedAccounts":
+                const numAllowedAccounts = await mycontract.methods.numAllowedAccounts().call();
+                console.log(`Number of allowed Acocunts: ${numAllowedAccounts}`);
+                break;
+            case "activate":
+                console.log(`Sending a transaction from account ${myAccount.address} to activate validator ${argv.validator}`);
+                receipt = await mycontract.methods.activate(getHex(argv.validator, 40, true, "validator")).send({from: myAccount.address});
+                status = receipt.status ? "Success" : "Failed";
```

Comment:
If the tx failed, we would  never receive the event we are waiting for below, would we? Also, in case of failure it would be nice to `exit(-1)`. And log to stderr as a stretch goal :-)

Same for all the modifying transactions.

Maybe this could be handled once for all transactions?

Reply:
All the functions, if we are not achieving what we want to achieve, throw an exception. In that case I changed to writing to stderr now!